hail/src/main/resources/include/hail/Utils.h

```diff
@@ -129,4 +133,20 @@ class ArrayLoadImpl : public BaseArrayImpl<elem_required, elem_size, elem_align>
   }
 };
 
+struct FatalError: public std::exception {
+  private:
+    static constexpr int max_error_len = 512;
+    int errno_;
+    char error_msg[max_error_len];
+  public:
+    virtual const char * what() const throw() { return (std::string("FatalError ") + std::to_string(errno_) + ": " + error_msg).c_str(); }
+    virtual ~FatalError() throw() {}
+    explicit FatalError(int errno, const char * fmtstring, ...) : std::exception(), errno_(errno) {
```

Comment:
You can't call this errno, in general that's a macro that expands to reference thread-local storage for the error number.

Follow up on the conversation in another comment, re: error numbers.  In addition to grep'ing, it can also be useful to have error numbers for a few other reasons, none of which we use now: it can help with localization, they can be used to disable certain errors/warnings, and they can be used anchors in documentation if you document individual errors.  I think you should just delete the error number.

Also, I'd make the error message a string and format it with a buffer in the constructor.  Also, I'd make the buffer a bit larger (8K?), although we should replace this with fmt soon.

Reply:
👌